arrow-libs/fx/arrow-collectors/src/commonMain/kotlin/arrow/collectors/Collect.kt

```diff
@@ -115,22 +114,20 @@ internal suspend fun <A, T, R> Flow<T>.collectI(
   concurrency: Int = DEFAULT_CONCURRENCY,
 ): R {
   var accumulator: A
-  val started = this.onStart { accumulator = collector.supply() }
-  val continued = when {
-    Characteristics.CONCURRENT in collector.characteristics -> when {
-      Characteristics.UNORDERED in collector.characteristics ->
-        started.parMapUnordered(concurrency) { collector.accumulate(accumulator, it) }
 
-      else ->
-        started.parMap(concurrency) { collector.accumulate(accumulator, it) }
-    }
+  val started = this.onStart { accumulator = collector.supply() }
 
+  val continued = when {
+    collector.has(Characteristics.CONCURRENT, Characteristics.UNORDERED) ->
```

Comment:
I'd prefer
```suggestion
    Characteristics.CONCURRENT in collector && Characteristics.UNORDERED in collector ->
```
Where we define an `operator fun CollectorI.contains(x: Characteristics) = x in characteristics`

Reply:
I am not 100% sure... `collector` itself does not have the characteristics, this is why I decided not to use something like `contains`. But if you strongly prefer that style, I'm happy to change it.